File: 199-binary-tree-right-side-view/199-binary-tree-right-side-view.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public List<Integer> rightSideView(TreeNode root) {
        if (root == null){
            return new ArrayList<Integer>();
        }
        ArrayDeque<TreeNode> queue = new ArrayDeque();
        List<Integer> ans = new ArrayList<>();
        queue.addLast(root);
        
        while (!queue.isEmpty()){
            int n = queue.size();
            TreeNode cur = null;
            
            for (int i = 0; i < n; ++i){
                cur = queue.pollFirst();
                if (cur.left != null){
                    queue.addLast(cur.left);
                }
                if (cur.right != null){
                    queue.addLast(cur.right);
                }
            }
            ans.add(cur.val);
        }
        return ans;
    }
}

File: 199-binary-tree-right-side-view/199-binary-tree-right-side-view.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def rightSideView(self, root: Optional[TreeNode]) -> List[int]:
        def dfs(rt, level):
            nonlocal result
            if not rt:
                return
            
            if len(result) == level:
                result.append(rt.val)
            
            dfs(rt.right, level + 1)
            dfs(rt.left, level + 1)
        
        result = []
        dfs(root, 0)
        return result
        